src/zope/deferredimport/deferredmodule.py

```diff
@@ -64,11 +64,15 @@ def __init__(self, module):
 
     def __getattr__(self, name):
         try:
-            get = self.__deferred_definitions__.pop(name)
+            get = self.__deferred_definitions__[name]
         except KeyError:
             raise AttributeError(name)
         v = get.get()
         setattr(self, name, v)
+        try:
+            del self.__deferred_definitions__[name]
+        except KeyError:
```

Comment:
This line and the following is uncovered in tests. If it is hard to write a test for them please mark them with `# pragma: no cover` to bring back coverage to 100 %.

Reply:
Or you could rewrite it as

```python
self.__deferred_definitions__.pop(name, None)
```

to avoid branching.

Reply:
Done.

I did not follow @mgedmin 's suggestion because the explicit `del` seems clearer to me.